featurebyte/session/bigquery.py

```diff
@@ -0,0 +1,584 @@
+        """
+        schema = None
+        if cursor.description:
+            schema = self._get_schema_from_cursor(cursor)
+
+        if schema:
+            records = cursor.fetchall()
+            if not records:
+                # return empty table to ensure correct schema is returned
+                return pa.record_batch([[]] * len(schema), schema=schema)
+            table = pa.table(list(zip(*[list(i.values()) for i in records])), schema=schema)
+            return table.to_pandas()
+        return None
+
+    async def fetch_query_stream_impl(self, cursor: Any) -> AsyncGenerator[pa.RecordBatch, None]:
+        schema = None
+        if cursor.description:
+            schema = self._get_schema_from_cursor(cursor)
+
+        if schema:
+            # fetch results in batches
+            while True:
+                records = cursor.fetchmany(BIGQUERY_BATCH_FETCH_SIZE)
+                if not records:
+                    # return empty table to ensure correct schema is returned
+                    yield pa.record_batch([[]] * len(schema), schema=schema)
+                    break
+                table = pa.table(list(zip(*[list(i.values()) for i in records])), schema=schema)
+                for batch in table.to_batches():
+                    yield batch
+
+    async def register_table(self, table_name: str, dataframe: pd.DataFrame) -> None:
+        self._client.load_table_from_dataframe(
+            dataframe, f"`{self.project_name}`.`{self.dataset_name}`.`{table_name}`"
+        )
+
+    async def list_table_schema(
+        self,
+        table_name: str | None,
+        database_name: str | None = None,
+        schema_name: str | None = None,
+        timeout: float = INTERACTIVE_SESSION_TIMEOUT_SECONDS,
+    ) -> OrderedDict[str, ColumnSpecWithDescription]:
+        assert table_name is not None
+        table_ref = TableReference(
+            dataset_ref=DatasetReference(project=self.project_name, dataset_id=self.dataset_name),
+            table_id=table_name,
+        )
+        table = self._client.get_table(table_ref)
+        column_name_type_map = collections.OrderedDict()
+        if table.schema is not None:
+            for field in table.schema:
+                dtype = self._convert_to_internal_variable_type(
+                    bigquery_typecode=field.field_type, scale=field.scale
+                )
+                column_name_type_map[field.name] = ColumnSpecWithDescription(
+                    name=field.name,
+                    dtype=dtype,
+                    description=field.description or None,
+                )
+
+        return column_name_type_map
+
+    async def get_table_details(
+        self,
+        table_name: str,
+        database_name: str | None = None,
+        schema_name: str | None = None,
+    ) -> TableDetails:
+        query = (
+            f'SELECT * FROM "{database_name}"."INFORMATION_SCHEMA"."TABLES" WHERE '
```

Comment:
Would it be better to only select the specific columns that we need in this query?

Reply:
This is actually from the snowflake implementation (and the functions below). Will work on the actual implementation and add test coverage